index/target_parser.hh

```diff
@@ -29,6 +29,8 @@ struct target_parser {
 
     static sstring get_target_column_name_from_string(const sstring& targets);
 
+    static sstring get_base_column_name_from_target(const sstring& target);
```

Comment:
I don't know what a "base column name" means and how it differs from "target column name". As https://github.com/scylladb/scylladb/pull/10841/commits/2c244c6e09c4693a47eadded930651599f8145d1 explains, we have three options now - a single column name (maybe quoted), a string(columname) (e.g., keys(colname), and a JSON (for "local index"). Which of these cases does this new function support and why?

Reply:
Maybe you are aiming for something like the existing function index_target::column_name_from_target_string()?

Reply:
Yes, exactly this. Thanks!